### PR TITLE
new(docker,scripts): port all docker images to be multiarch ready

### LINKED ISSUE #1589: Support arm64 architecture in dockerhub
**Motivation**
Tried to run Falco in a arm64 AWS EC2 AMI amazon-eks-arm64-node-*. Getting the error:

`standard_init_linux.go:219: exec user process caused: exec format error`

Realise that the docker hub image only has amd64 architecture, which means it fails with a multi architecture kubernetes cluster.

**Feature**
Create a multi architecture docker image.

### LINKED ISSUE #1813: Multi-arch docker image request
**Motivation**

 Our deployment contains a mix of x86 and arm.  Having a single multi-arch Docker image to deploy using helm on both x86 and arm platforms would be really helpful and useful feature for our team.  

**Feature**

Buildx (https://docs.docker.com/buildx/working-with-buildx/ ) - This is the tool used by use for internal images and a possible option.

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

> If contributing rules or changes to rules, please make sure to also uncomment one of the following line:

> /kind rule-update

> /kind rule-create

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area build

> /area engine

> /area rules

> /area tests

> /area proposals

**What this PR does / why we need it**:

New dockerfiles are multiplatform ready: they can be pushed with `docker buildx` for various architectures.
Moreover, updated falco-driver-loader to support multiple architectures.

**Which issue(s) this PR fixes**:

Fixes #1813
Fixes #1589

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -389,74 +389,80 @@ jobs:
   # Publish docker packages
   "publish/docker-dev":
     docker:
-      - image: docker:stable
+      - image: cimg/base:stable
```
Comment (FedeDP) [inline .circleci/config.yml:392]: Switched to cimg/base:stable as it already provides `docker buildx` command for us, when a 20.10.x version of docker is requested.

File: scripts/falco-driver-loader [hunk 2]
```diff
@@ -201,7 +201,12 @@ load_kernel_module_download() {
 	local FALCO_KERNEL_MODULE_FILENAME="${DRIVER_NAME}_${TARGET_ID}_${KERNEL_RELEASE}_${KERNEL_VERSION}.ko"
 
 	local URL
-	URL=$(echo "${DRIVERS_REPO}/${DRIVER_VERSION}/${FALCO_KERNEL_MODULE_FILENAME}" | sed s/+/%2B/g)
+	if [ "${ARCH}" == "x86_64" ]; then
+		# x86_64 still uses root
+		URL=$(echo "${DRIVERS_REPO}/${DRIVER_VERSION}/${FALCO_KERNEL_MODULE_FILENAME}" | sed s/+/%2B/g)
```
Comment (FedeDP) [inline scripts/falco-driver-loader:206]: We will leave x86_64 drivers in their root folder, where they are now, to retain backward compatibility.
New archs will instead go below $arch subfolder.

**Comment by FedeDP** [comment]
I just ported `publish-rpm` and `publish-deb` scripts to be multiarch. 
They must be now called once with list of deb files (one for each supported arch).

Publish-rpm was locally tested, while publish-deb was tested on same docker image used by circleCI: `docker.io/debian:stable`, to double check the added script dependency (`dpkg`).

Of course, in both test, i disabled aws s3/cloudfron download and sync.

Publish-deb was the harder one; basically, i fixed the publsih-deb script and then tested adding the new local repo to apt sources, and double checked that apt could actually update and install from the repo.
The repo tree must be equal to https://download.falco.org/?prefix=packages/deb/.
```
root@60afe6470047:/tmp/falco-deb# ls
dists  stable

ls dists/stable/
Release  Release.gpg  main

root@60afe6470047:/tmp/falco-deb# cat dists/stable/Release
Architectures: amd64,arm64
Codename: stable
Components: main
Date: Fri, 13 May 2022 09:21:52 +0000
Label: Falco
Origin: Falco
Suite: stable
MD5Sum:
5825c4e96c8acfb055ba545635c63b36              140 Release
affb84c8f074475175ae46ba9b158af7             1205 main/binary-amd64/Packages
bebf147595828eb82fadc8f7bdea68d7              824 main/binary-amd64/Packages.bz2
077fde40e75311196557166ce81d4fa3              783 main/binary-amd64/Packages.gz
0fd7b90883437a343949b9c71063e6de             1206 main/binary-arm64/Packages
d4d22eb0f42c88615118eb49e1777fd4              820 main/binary-arm64/Packages.bz2
0e8adb1fb50a24c5ee5b2abefcc43328              784 main/binary-arm64/Packages.gz
[more...]

root@60afe6470047:/tmp/falco-deb# cat dists/stable/main/binary-arm64/Packages
Package: geoip-bin
Architecture: arm64
Version: 1.6.12-8
[more...]

root@60afe6470047:/tmp/falco-deb# cat dists/stable/main/binary-amd64/Packages
Package: geoip-bin
Architecture: amd64
Version: 1.6.12-8
[more...]

root@60afe6470047:/tmp/falco-deb# ls stable/
geoip-bin_1.6.12-8_amd64.deb  geoip-bin_1.6.12-8_amd64.deb.asc  geoip-bin_1.6.12-8_arm64.deb  geoip-bin_1.6.12-8_arm64.deb.asc

root@60afe6470047:/tmp/falco-deb# apt-cache showpkg geoip-bin
Package: geoip-bin
Versions:
1.6.12-8 (/var/lib/apt/lists/_tmp_falco-deb_dists_stable_main_binary-amd64_Packages.lz4)
```

**Review by leogr — COMMENTED** [review]
This is spectacular, I just need to find some time to test it deeply.

Thank you! :hugs:

**Comment by leogr** [comment]
I think we don't have enough time to test this. Moving to the next milestone.
/milestone 0.33.0

**Comment by jasondellaluce** [comment]
@FedeDP, should we link this to https://github.com/falcosecurity/falco/issues/1813 with __Fixes#__ if that's gonna be closed once this will get merged?

**Comment by FedeDP** [comment]
You are right jason! Thanks!

**Comment by jasondellaluce** [comment]
This one too I think! https://github.com/falcosecurity/falco/issues/1589

**Review by jasondellaluce — APPROVED** [review]
/approve

This is a big change, but since we just released Falco 0.32 I think this is the right time to test it! Great work Fede!

**Review by leogr — APPROVED** [review]

**Pull request merged**